contracts/SpokePoolVerifier.sol

```diff
@@ -23,41 +23,51 @@ contract SpokePoolVerifier {
      * This contract should only be used for native token deposits, as this problem only exists for native tokens.
      * @param spokePool Address of the SpokePool contract that the user is intending to call.
      * @param recipient Address to receive funds at on destination chain.
-     * @param originToken Token to lock into this contract to initiate deposit.
-     * @param amount Amount of tokens to deposit. Will be amount of tokens to receive less fees.
+     * @param inputToken Token to lock into this contract to initiate deposit.
+     * @param inputAmount Amount of tokens to deposit.
+     * @param outputAmount Amount of tokens to receive on destination chain.
      * @param destinationChainId Denotes network where user will receive funds from SpokePool by a relayer.
-     * @param relayerFeePct % of deposit amount taken out to incentivize a fast relayer.
      * @param quoteTimestamp Timestamp used by relayers to compute this deposit's realizedLPFeePct which is paid
      * to LP pool on HubPool.
      * @param message Arbitrary data that can be used to pass additional information to the recipient along with the tokens.
      * Note: this is intended to be used to pass along instructions for how a contract should use or allocate the tokens.
-     * @param maxCount used to protect the depositor from frontrunning to guarantee their quote remains valid.
-     * Note: this is intended to be used to pass along instructions for how a contract should use or allocate the tokens.
+     * @param exclusiveRelayer Address of the relayer who has exclusive rights to fill this deposit. Can be set to
+     * 0x0 if no exclusivity period is desired. If so, then must set exclusivityDeadline to 0.
+     * @param exclusivityDeadline Timestamp after which any relayer can fill this deposit. Must set
+     * to 0 if exclusiveRelayer is set to 0x0, and vice versa.
+     * @param fillDeadline Timestamp after which this deposit can no longer be filled.
      */
     function deposit(
-        SpokePoolInterface spokePool,
+        V3SpokePoolInterface spokePool,
         address recipient,
-        address originToken,
-        uint256 amount,
+        address inputToken,
+        uint256 inputAmount,
+        uint256 outputAmount,
         uint256 destinationChainId,
-        int64 relayerFeePct,
         uint32 quoteTimestamp,
         bytes memory message,
-        uint256 maxCount
+        address exclusiveRelayer,
+        uint32 exclusivityDeadline,
```

Comment:
Should we reorder the arguments so the calldata is smaller? Might also be nice to use the same structure as `depositV3` just for organization purposes.